core/test_network/src/unique_port.rs

```diff
@@ -0,0 +1,65 @@
+#![allow(clippy::restriction)]
+
+use once_cell::sync::Lazy;
+use std::net::{Ipv4Addr, SocketAddrV4, TcpListener};
+use std::ops::Range;
+use std::sync::Mutex;
+
+static PORT_IDX: Lazy<Mutex<u16>> = Lazy::new(|| Mutex::new(1000));
+
+#[macro_export]
+macro_rules! generate_offset {
+    () => {{
+        use std::collections::hash_map::DefaultHasher;
+        use std::hash::{Hash, Hasher};
+
+        fn f() {}
+        fn type_name_of<T>(_: T) -> &'static str {
+            std::any::type_name::<T>()
+        }
+        let name = type_name_of(f);
+        let name = &name[..name.len() - 3];
+        let mut hasher = DefaultHasher::new();
+        name.hash(&mut hasher);
+        1000 + (hasher.finish() % ((u16::MAX - 1000) as u64)) as u16
+    }};
+}
+
+pub fn set_offset(offset: u16) -> Result<(), String> {
+    let mut port_idx = PORT_IDX
+        .lock()
+        .map_err(|_| "Failed to aquire the lock".to_owned())?;
+    *port_idx = offset;
+
+    Ok(())
+}
+
+/// Returns a free unique local port. Every time a call to this function during one run should
+/// return a unique address.
+///
+/// # Examples
+/// ```
+/// use unique_port::get_unique_free_port;
+///
+/// let port_1 = get_unique_free_port().unwrap();
+/// let port_2 = get_unique_free_port().unwrap();
+/// assert_ne!(port_1, port_2);
+/// ```
+pub fn get_unique_free_port() -> Result<u16, String> {
+    let mut port_idx = PORT_IDX
```

Comment:
Do we still need this to be a lock?

Reply:
Why is it a `Mutex`?

Reply:
I think we need it to be thread-safe. Not only because it won't work with `cargo test` without it, but to safely generate ports in multi-threading context.

For the `Mutex`, as I've written in the description,  it's just a copypasta of [unique_port](https://github.com/soramitsu/iroha2-unique_port) implementation. And I'd rather provide a PR with these changes there.